### PR TITLE
Marketing Phone

### PR DESCRIPTION
**Pull request by Trilbyspaceclone**
NEW ceo buff!!!!
New now gets a fancy phone that allows them to unlock 1 (1!!!) station for free
The fancy new phone also now can be charged with ewallets to direct give a station of the CEOs choice funding

### CODE REVIEW

File: code/game/objects/items/oddities_faction.dm [hunk 1]
```diff
@@ -727,3 +727,128 @@ No more of that.
 					"potato",
 					"rice")]
 				S.harvest(get_turf(src),0,0,1)
+
+
+
+/obj/item/direct_line
+	name = "direct phone line"
+	desc = "A powerful direct callline to local ships in the area, as well as having a slot to charge the phone with cashcards to directly fund a station of the CEO's choice."
```
Comment (ThePainkiller) [inline code/game/objects/items/oddities_faction.dm:735]:
```suggestion
	desc = "A powerful direct hotline to local ships in the area, as well as having a slot to charge the phone with cashcards to directly fund a station of the CEO's choice."
```

File: code/game/objects/items/oddities_faction.dm [hunk 2]
```diff
@@ -727,3 +727,128 @@ No more of that.
...
+	icon = 'icons/obj/items.dmi'
+	icon_state = "phone"
+	flags = CONDUCT
+	force = WEAPON_FORCE_HARMLESS
+	throwforce = WEAPON_FORCE_HARMLESS
+	throw_speed = 1
+	throw_range = 4
+	w_class = ITEM_SIZE_SMALL
+	price_tag = 4000
+	attack_verb = list("cold calls", "freezes assets")
```
Comment (ThePainkiller) [inline code/game/objects/items/oddities_faction.dm:745]:
```suggestion
	attack_verb = list("liquidates", "divests", "outsources", "lays off", "downsizes")
```

File: code/game/objects/items/oddities_faction.dm [hunk 3]
```diff
@@ -727,3 +727,128 @@ No more of that.
...
+
+		if("Funding Operations")
+			if(credits_to_fund)
+				var/list/stations_list_for_fund = list()
+				for(var/stations_to_fund in SStrade.discovered_stations)
+					var/datum/trade_station/O = stations_to_fund
+					stations_list_for_fund += O
+
+				if(!stations_list_for_fund)
+					to_chat(user, SPAN_NOTICE("When you ring the line, you hear an annoyed voice say ''I'm prepared to close this deal quickly.'' and then the line goes silent. Seems like their is no stations willing to trade with the colony."))
```
Comment (ThePainkiller) [inline code/game/objects/items/oddities_faction.dm:777]:
```suggestion
					to_chat(user, SPAN_NOTICE("When you ring the line, you hear an annoyed voice say \"I'm prepared to close this deal quickly\", followed by a silent dial tone. Seems like there are no stations willing to trade with the colony."))
```

File: code/game/objects/items/oddities_faction.dm [hunk 4]
```diff
@@ -727,3 +727,128 @@ No more of that.
...
+					return
+
+				var/datum/trade_station/station_choice = input(user, "What Station do you want to force sales with you?") as null|anything in stations_list_for_fund
```
Comment (ThePainkiller) [inline code/game/objects/items/oddities_faction.dm:780]:
```suggestion
				var/datum/trade_station/station_choice = input(user, "What Station do you want to force sales with?") as null|anything in stations_list_for_fund
```

File: code/game/objects/items/oddities_faction.dm [hunk 5]
```diff
@@ -727,3 +727,128 @@ No more of that.
...
+				if(!station_choice)	// user can cancel
+					to_chat(user, SPAN_NOTICE("You quickly hang up the phone."))
+					return
+				station_choice.wealth += credits_to_fund
+				to_chat(user, SPAN_NOTICE("Before you work out a deal to transfer the funding you hear ''[pick("You rang?","I was planning on running into you.","This caller always collects.","I'm one smooth operator.","I always make my calls direct.")]'' and then the line goes silent, as the counter on the side for funding slowly ticks down..."))
+				credits_to_fund = 0
+			else
+				to_chat(user, SPAN_NOTICE("When you ring the line, you hear an annoyed voice say ''I'm prepared to close this deal quickly.'' and then the line goes silent. Its likely do to the phone not having any money to fund to a station."))
```
Comment (ThePainkiller) [inline code/game/objects/items/oddities_faction.dm:788]:
```suggestion
				to_chat(user, SPAN_NOTICE("When you ring the line, you hear an annoyed voice say \"Sorry, we cannot give credit. Call back when you're a little mmm.... richer.\" followed by a silent dial tone. It's likely due to the phone not having enough money to directly fund a station."))
```

File: code/game/objects/items/oddities_faction.dm [hunk 6]
```diff
@@ -727,3 +727,128 @@ No more of that.
...
+				anti_cheat = TRUE
+				var/list/stations_list_for_phone = list()
+				for(var/stations_to_unlock in SStrade.all_stations)
+					if(stations_to_unlock in SStrade.discovered_stations)
+						continue
+					var/datum/trade_station/O = stations_to_unlock
+					stations_list_for_phone += O
+
+				if(!stations_list_for_phone)
+					to_chat(user, SPAN_NOTICE("When you ring the line, you hear an annoyed voice say ''I'm prepared to close this deal quickly.'' and then the line goes silent. Looks like they are not in any mood to do more Aggressive Sales Market"))
```
Comment (ThePainkiller) [inline code/game/objects/items/oddities_faction.dm:824]:
```suggestion
					to_chat(user, SPAN_NOTICE("When you ring the line, you hear an annoyed voice say /"I'm prepared to close this deal quickly./" followed by a silent dial tone. Looks like they are not in any mood to do more Aggressive Sales Market."))
```

File: code/game/objects/items/oddities_faction.dm [hunk 7]
```diff
@@ -727,3 +727,128 @@ No more of that.
...
+				var/station_choice = input(user, "What Station do you want to force sales with you?") as null|anything in stations_list_for_phone
+				if(!station_choice)	// user can cancel
+					anti_cheat = FALSE
+					to_chat(user, SPAN_NOTICE("You quickly hang up the phone."))
+					return
+				SStrade.discovered_stations |= station_choice
+				unlocks_left -= 1
+				anti_cheat = FALSE
+			else
+				to_chat(user, SPAN_NOTICE("When you ring the line, you hear an annoyed voice say ''I'm prepared to close this deal quickly.'' and then the line goes silent. Looks like they are not in any mood to do more Aggressive Sales Market"))
```
Comment (ThePainkiller) [inline code/game/objects/items/oddities_faction.dm:836]:
```suggestion
				to_chat(user, SPAN_NOTICE("When you ring the line, you hear an annoyed voice say /"I'm prepared to close this deal quickly./" followed by a silent dial tone. Looks like they are not in any mood to do more Aggressive Sales Market."))
```

File: code/game/objects/items/oddities_faction.dm [hunk 8]
```diff
@@ -727,3 +727,128 @@ No more of that.
...
+
+		else
+			return
+
+
+/obj/item/direct_line/attackby(obj/item/I, mob/living/user, params)
+	if(istype(I, /obj/item/spacecash/ewallet))
+		var/obj/item/spacecash/ewallet/C = I
+		if(C.worth <= 0)
+			to_chat(user, SPAN_NOTICE("When you try to transfer the ewallet you hear on the line ''I'm prepared to close this deal quickly.'' and then the line goes silent..."))
```
Comment (ThePainkiller) [inline code/game/objects/items/oddities_faction.dm:847]:
```suggestion
			to_chat(user, SPAN_NOTICE("When you try to transfer the e-wallet funds to the line, you hear \"Sorry, we cannot give credit. Call back when you're a little mmm.... richer.\" followed by a silent dial tone. It's likely due to not having enough funds on it..."))
```

**Review by ThePainkiller — CHANGES_REQUESTED** [review]
Trimming the typos

File: code/game/objects/items/oddities_faction.dm [hunk 9]
```diff
@@ -727,3 +727,128 @@ No more of that.
...
+				anti_cheat = TRUE
+				var/list/stations_list_for_phone = list()
+				for(var/stations_to_unlock in SStrade.all_stations)
+					if(stations_to_unlock in SStrade.discovered_stations)
+						continue
+					var/datum/trade_station/O = stations_to_unlock
+					stations_list_for_phone += O
+
+				if(!stations_list_for_phone)
+					to_chat(user, SPAN_NOTICE("When you ring the line, you hear an annoyed voice say /"I'm prepared to close this deal quickly./" followed by a silent dial tone. Looks like they are not in any mood to do more Aggressive Sales Market."))
```
Comment (Trilbyspaceclone) [inline code/game/objects/items/oddities_faction.dm:824]:
```suggestion
					to_chat(user, SPAN_NOTICE("When you ring the line, you hear an annoyed voice say \"I'm prepared to close this deal quickly.\" followed by a silent dial tone. Looks like they are not in any mood to do more Aggressive Sales Market."))
```

File: code/game/objects/items/oddities_faction.dm [hunk 10]
```diff
@@ -727,3 +727,128 @@ No more of that.
...
+				var/station_choice = input(user, "What Station do you want to force sales with you?") as null|anything in stations_list_for_phone
+				if(!station_choice)	// user can cancel
+					anti_cheat = FALSE
+					to_chat(user, SPAN_NOTICE("You quickly hang up the phone."))
+					return
+				SStrade.discovered_stations |= station_choice
+				unlocks_left -= 1
+				anti_cheat = FALSE
+			else
+				to_chat(user, SPAN_NOTICE("When you ring the line, you hear an annoyed voice say /"I'm prepared to close this deal quickly./" followed by a silent dial tone. Looks like they are not in any mood to do more Aggressive Sales Market."))
```
Comment (Trilbyspaceclone) [inline code/game/objects/items/oddities_faction.dm:836]:
```suggestion
				to_chat(user, SPAN_NOTICE("When you ring the line, you hear an annoyed voice say \"I'm prepared to close this deal quickly.\" followed by a silent dial tone. Looks like they are not in any mood to do more Aggressive Sales Market."))
```

**Pull request merged**